src/Extensions/AzureBlobStorage/PostConfigureAzureBlobEgressProviderOptions.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.Extension.Common;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Diagnostics.Monitoring.AzureBlobStorage
+{
+    internal sealed class PostConfigureAzureBlobEgressProviderOptions :
+        IPostConfigureOptions<AzureBlobEgressProviderOptions>
+    {
+        private readonly ILogger _logger;
+        private readonly EgressProperties _properties;
+
+        public PostConfigureAzureBlobEgressProviderOptions(ILogger<AzureBlobEgressProvider> logger, EgressProperties properties)
+        {
+            _properties = properties;
+            _logger = logger;
+        }
+
+        public void PostConfigure(string name, AzureBlobEgressProviderOptions options)
+        {
+            // If account key was not provided but the name was provided,
+            // lookup the account key property value from EgressOptions.Properties
```

Comment:
nit: I realize this comment was already out of date, but let's fix the reference to `EgressOptions.Properties` for all three of these.